src/utils/hooks/useFetch.js

```diff
@@ -0,0 +1,48 @@
+import { useState, useEffect } from 'react';
+
+const key = process.env.REACT_APP_API_KEY;
+
+const baseUrl = 'https://www.googleapis.com/youtube/v3/search?';
+
+export default function useFetch(query, relatedVideos = false) {
```

Comment:
Since this function is video-specific, a more appropriate name could be `useVideoAPI` or something like that